plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1170,7 +1173,29 @@ else if (avroSchemaLiteral != null) {
         // Partition Projection specific properties
         tableProperties.putAll(partitionProjectionService.getPartitionProjectionHiveTableProperties(tableMetadata));
 
-        return tableProperties.buildOrThrow();
+        Map<String, String> baseProperties = tableProperties.buildOrThrow();
+
+        // Extra properties
+        Map<String, String> extraProperties = getExtraProperties(tableMetadata.getProperties())
+                .orElseGet(ImmutableMap::of);
+
+        Set<String> existingProperties = ImmutableSet.<String>builder()
+                .addAll(baseProperties.keySet())
+                .addAll(STATS_PROPERTIES)
+                .build();
+
+        Set<String> overlappingProperty = Sets.intersection(existingProperties, extraProperties.keySet());
+
+        if (!overlappingProperty.isEmpty()) {
+            throw new TrinoException(
+                    INVALID_TABLE_PROPERTY,
+                    "Invalid keys in extra_properties: %s".formatted(String.join(" ,", overlappingProperty)));
```

Comment:
`join(" ,"` or `join(", "` ?

wouldn't overlappingProperty's toString be enough? `.... in extra_properties: " + overlappingProperty`

Invalid -> Illegal